fluent-bundle/src/bundle.ts

```diff
@@ -196,7 +196,7 @@ export class FluentBundle {
       let value = resolveComplexPattern(scope, pattern);
       return value.toString(scope);
     } catch (err) {
-      if (scope.errors) {
+      if (scope.errors && err instanceof Error) {
```

Comment:
Question: Does this risk being a behavior change? It could affect users. I don't know how errors move through the system and if we can assume that it will be properly an instance of Error. This applies to the other changes elsewhere. I'm assuming this change is addressing is a new type error.

Reply:
I traced through all of the `throw` statements in the repo, and there's no reason to suspect that anything other than an `Error` or one of its subclasses might be thrown.

This change is due to a change of defaults in TypeScript 4.4, specifically this: https://devblogs.microsoft.com/typescript/announcing-typescript-4-4/#use-unknown-catch-variables